compat_test.go

```diff
@@ -189,6 +164,484 @@ func (*suite) TestBind(c *gc.C) {
 	})
 }
 
+type conditionTest struct {
+	conditions    map[string]bool
+	expectFailure []mcompat.Implementation
+	expectErr     string
+}
+
+var verifyTests = []struct {
+	about      string
+	macaroons  []macaroonSpec
+	conditions []conditionTest
+}{{
+	about: "single third party caveat without discharge",
+	macaroons: []macaroonSpec{{
+		rootKey: "root-key",
+		id:      "root-id",
+		caveats: []caveat{{
+			condition: "wonderful",
+		}, {
+			condition: "bob-is-great",
+			location:  "bob",
+			rootKey:   "bob-caveat-root-key",
+		}},
+	}},
+	conditions: []conditionTest{{
+		conditions: map[string]bool{
+			"wonderful": true,
+		},
+		expectErr: `cannot find discharge macaroon for caveat "bob-is-great"`,
+	}},
+}, {
+	about: "single third party caveat with discharge",
```

Comment:
Each set of caveats is tested with a number of conditions, so the "about" comment refers to the macaroons themselves rather than the conditions.
In this particular case, there are two verifications in the condition tests, one of which is expected to pass, the other fail.

FWIW this entire table is taken verbatim from gopkg.in/macaroon.v1/macaroon_test.go, so if we do decide to change something, it's probably good to change it there too.